app/src/main/java/com/example/parkingsystemjava/mvp/presenter/ReservationPresenter.java

```diff
@@ -34,14 +34,34 @@ public void setReservationCalendar(Calendar date, boolean isStartDate) {
 
     @Override
     public void buttonSaveReservation() {
-        model.saveReservation(view.getParkingLots(), view.getUserPassword());
-        Reservation reservation = model.getReservation(view.getParkingLots(), view.getUserPassword());
-        view.showMessageOfSavedStatement(reservation);
-        view.dismissActivity();
+        Reservation reservation = model.createReservation(view.getParkingLots(), view.getUserPassword());
+        model.existsOverlap(reservation);
+        if (!model.isOverlap()) {
+            if (check(reservation)) {
+                model.saveReservation();
+                view.showMessageOfSavedStatement(reservation);
+                view.dismissActivity();
+            } else {
+                view.showMessageOfError();
+            }
+        } else {
+            view.showMessageOfOverlapReservation();
+        }
     }
 
     @Override
     public void buttonCancelReservation() {
         view.dismissActivity();
     }
+
+    @Override
+    public boolean check(Reservation reservation) {
+        if (reservation != null) {
+            if (reservation.getStartDate() != null || reservation.getEndDate() != null ||
+                    !reservation.getParkingLots().equals("") || !reservation.getUserPassword().equals("")) {
```

Comment:
same here, do not hardcode

Reply:
Done, changed with a constant in the ConstantUtils class